### PR TITLE
Use full Post schema to filter fields based on context

### PR DESCRIPTION
**Pull request by danielbachhuber**
See #894

### CODE REVIEW

File: lib/endpoints/class-wp-json-attachments-controller.php [hunk 1]
```diff
@@ -167,6 +167,9 @@ public function prepare_item_for_response( $post, $request ) {
 		    $response['media_details']['sizes'] = new stdClass;
 		}
 
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
```
Comment (joehoyle) [inline lib/endpoints/class-wp-json-attachments-controller.php]: @danielbachhuber can this be handled by a default param in the `args` in the route registration?

File: lib/endpoints/class-wp-json-posts-controller.php [hunk 2]
```diff
@@ -964,60 +964,53 @@ public function prepare_item_for_response( $post, $request ) {
...
+
 			$data['content'] = array(
+				'raw'      => $post->post_content,
 				'rendered' => apply_filters( 'the_content', $post->post_content ),
 			);
-			if ( 'edit' === $request['context'] ) {
-				$data['content']['raw'] = $post->post_content;
+
+			if ( ! empty( $post->post_password ) ) {
+				$_COOKIE['wp-postpass_' . COOKIEHASH] = '';
```
Comment (joehoyle) [inline lib/endpoints/class-wp-json-posts-controller.php]: Can we get a comment in here to explain this crazyness?

File: lib/endpoints/class-wp-json-posts-controller.php [hunk 3]
```diff
@@ -1080,6 +1073,9 @@ public function prepare_item_for_response( $post, $request ) {
 			) );
 		}
 
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
```
Comment (joehoyle) [inline lib/endpoints/class-wp-json-posts-controller.php]: Again, seems like this should be a registered arg with a default value?

File: lib/endpoints/class-wp-json-attachments-controller.php (hunk 1 above)
Comment (danielbachhuber) [inline lib/endpoints/class-wp-json-attachments-controller.php]: > can this be handled by a default param in the args in the route registration?

Maybe. But, conceivably, another controller can call this directly and omit `context`

Comment (joehoyle) [inline lib/endpoints/class-wp-json-attachments-controller.php]: Fair enough

File: lib/endpoints/class-wp-json-posts-controller.php [hunk 4]
```diff
@@ -1198,24 +1191,49 @@ public function get_item_schema() {
...
+					'description' => 'A password to protect access to the post.',
+					'type'        => 'string',
+					'context'     => array( 'edit' ),
+				),
+				'slug'            => array(
+					'description' => 'An alphanumeric identifier for the object unique to its type.',
+					'type'        => 'string',
+					'context'     => array( 'view', 'edit' ),
+				),
+				'status'          => array(
```
Comment (joehoyle) [inline lib/endpoints/class-wp-json-posts-controller.php]: an we pass enums here?

File: lib/endpoints/class-wp-json-posts-controller.php (hunk 2 above)
Comment (danielbachhuber) [inline lib/endpoints/class-wp-json-posts-controller.php]: 8d53589

File: lib/endpoints/class-wp-json-posts-controller.php (hunk 4 above)
Comment (danielbachhuber) [inline lib/endpoints/class-wp-json-posts-controller.php]: dd05e31

**Pull request merged**